File: mediapicker/src/main/java/iam/thevoid/mediapicker/rxmediapicker/metrics/MemorySize.java
package iam.thevoid.mediapicker.rxmediapicker.metrics;

public class MemorySize {
    private long size;
    private SizeUnit unit;

    public MemorySize(int size, SizeUnit unit) {

        if (size < 0) {
            this.size = 0;
        }

        this.size = size;
        this.unit = unit;
    }

    public long getBytes() {
        return size * unit.getBytes();
    }
}

File: core/src/main/java/iam/thevoid/mediapicker/picker/metrics/MemorySize.kt
package iam.thevoid.mediapicker.picker.metrics

data class MemorySize(private val size: Int = -1, private val unit: SizeUnit = SizeUnit.BYTE) {

    val bytes: Long
        get() = size * unit.bytes

    val kiloBytes: Long
        get() = bytes / SizeUnit.KILOBYTE.bytes

}